test/bin/encore.js

```diff
@@ -213,4 +215,128 @@ module.exports = Encore.getWebpackConfig();
             done();
         });
     });
+
+    it('Run the webpack-dev-server successfully', (done) => {
+        testSetup.emptyTmpDir();
+        const testDir = testSetup.createTestAppDir();
+
+        fs.writeFileSync(
+            path.join(testDir, 'package.json'),
+            `{
+                "devDependencies": {
+                    "@symfony/webpack-encore": "*"
+                }
+            }`
+        );
+
+        fs.writeFileSync(
+            path.join(testDir, 'webpack.config.js'),
+            `
+const Encore = require('../../index.js');
+Encore
+    .enableSingleRuntimeChunk()
+    .setOutputPath('build/')
+    .setPublicPath('/build')
+    .addEntry('main', './js/no_require')
+;
+
+module.exports = Encore.getWebpackConfig();
+            `
+        );
+
+        const binPath = path.resolve(__dirname, '../', '../', 'bin', 'encore.js');
+        const abortController = new AbortController();
+        const node = spawn('node', [binPath, 'dev-server', `--context=${testDir}`], { cwd: testDir, env: { ...process.env, NO_COLOR: 'true' }, signal: abortController.signal });
+
+        let stdout = '';
+        let stderr = '';
+
+        node.stdout.on('data', (data) => {
+            stdout += data.toString();
+        });
+
+        node.stderr.on('data', (data) => {
+            stderr += data.toString();
+        });
+
+        node.on('error', (error) => {
+            if (error.name !== 'AbortError') {
+                throw new Error('Error executing encore', { cause: error });
+            }
+
+            expect(stdout).to.contain('Running webpack-dev-server ...');
+            expect(stdout).to.contain('Compiled successfully in');
+            expect(stdout).to.contain('webpack compiled successfully');
+
+            expect(stderr).to.contain('[webpack-dev-server] Project is running at:');
+            expect(stderr).to.contain('[webpack-dev-server] Loopback: http://localhost:8080/, http://127.0.0.1:8080/');
+            expect(stderr).to.contain('[webpack-dev-server] Content not from webpack is served from');
+
+            done();
+        });
+
+        setTimeout(() => {
+            abortController.abort();
+        }, 2500);
+    });
+
+    describe('Without webpack-dev-server installed', () => {
+        const webpackDevServerConstraint = require('../../package.json').devDependencies['webpack-dev-server'] || null;
+        if (!webpackDevServerConstraint) {
+            throw new Error('Missing "webpack-dev-server" as dev dependency in package.json.');
+        }
+
+        before(() => {
+            execSync('yarn remove webpack-dev-server --dev', { cwd: projectDir });
+        });
+
+        after(() => {
+            // Re-install webpack-dev-server and ensure the project is in a clean state
+            execSync(`yarn add webpack-dev-server@${webpackDevServerConstraint} --dev`, { cwd: projectDir });
```

Comment:
why not doing a `git checkout package.json` instead ?

Reply:
I think about it, but I don't wanted to checkout unrelated modifications by accident

Reply:
wouldn't this also apply to the lock file then ?

Is there a way to write such tests without changing the dev dependencies of encore ?

Reply:
> wouldn't this also apply to the lock file then ?

I'm not sure to understand, you mean `git checkout yarn.lock`? If yes, it won't help to restore remove/re-add `webpack-dev-server` dependency in `package.json`

> Is there a way to write such tests without changing the dev dependencies of encore ?

I first thought about writing E2E tests like the one for our "test apps":
- clone Encore
- install prod dependencies and pack Encore, ensure webpack-dev-server is not present
- create a new project (or reuse a test-app)
- write a bash assertion for `encore dev-server` 

But it's not really the goal of our actual "test apps", I also wanted to check lowest/fixed/highest dependencies, and I prefer to write tests in `test/*.js` files.

Reply:
I've changed to a `git checkout package.json yarn.lock` instead, it's more simple, and too bad if it cancels unrelated modifications.